### PR TITLE
Pom cleanup

### PR DESCRIPTION
**Pull request by costin-zaharia-sonarsource**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -92,6 +110,68 @@
...
+            <artifactId>stax-api</artifactId>
+          </exclusion>
+        </exclusions>
+        <scope>test</scope>
+      </dependency>
+      <dependency>
+        <groupId>org.codehaus.staxmate</groupId>
+        <artifactId>staxmate</artifactId>
+        <version>${staxmate.version}</version>
+        <exclusions>
```
Comment (antonioaversa) [inline pom.xml:159]: Are exclusions propagated from `depedencyManagement` to the module actually using the dependency (in this case `sonar-csharp-plugin`)?
Could you just double check the effective pom of the plugin, and confirm that `dependencies` report the exclusions?
I am mostly concerned about https://stackoverflow.com/questions/39276024/import-dependency-management-with-exclusion

File: sonar-vbnet-plugin/pom.xml [hunk 2]
```diff
@@ -50,7 +50,6 @@
     <dependency>
       <groupId>org.sonarsource.api.plugin</groupId>
       <artifactId>sonar-plugin-api</artifactId>
-      <version>${sonar.version}</version>
       <scope>provided</scope>
```
Comment (antonioaversa) [inline sonar-vbnet-plugin/pom.xml:53]: Can we have a comment explaining what is providing the `sonar-plugin-api` (I guess SonarQube/SonarCloud) and why we need the scope defined at this level?

As for the second question: I am not sure we need it here: the scope is already defined as `provided` in the `dependencyManagement` of the pom at root-level.

File: sonar-csharp-plugin/pom.xml [hunk 3]
```diff
@@ -50,7 +50,6 @@
     <dependency>
       <groupId>org.sonarsource.api.plugin</groupId>
       <artifactId>sonar-plugin-api</artifactId>
-      <version>${sonar.version}</version>
       <scope>provided</scope>
```
Comment (antonioaversa) [inline sonar-csharp-plugin/pom.xml:53]: Same as for comment below: can we have a comment explaining what is providing the sonar-plugin-api (I guess SonarQube/SonarCloud) and why we need the scope defined at this level?

File: pom.xml [hunk 4]
```diff
@@ -92,6 +110,68 @@
...
+      <dependency>
+        <groupId>org.junit.jupiter</groupId>
+        <artifactId>junit-jupiter-api</artifactId>
+        <version>${junit.version}</version>
+        <scope>test</scope>
+      </dependency>
+      <dependency>
+        <groupId>org.assertj</groupId>
+        <artifactId>assertj-core</artifactId>
+        <version>${assertj.core.version}</version>
```
Comment (antonioaversa) [inline pom.xml:122]: I don't think the Maven Property `assertj.core.version` is needed anymore: now that the version is only defined at `dependencyManagement` level, there is a single usage of the variable, so we can inline it. 

The same seems to apply to 
- commons.lang.version
- junit.version
- mockito.version
- stax2.api.version
- staxmate.version

**Review by antonioaversa — APPROVED** [review]
LGTM, nice factorization of versioning dependencies, that greatly helps with version management!
I left a few minor suggestions.

File: pom.xml (hunk 1 above)
Comment (costin-zaharia-sonarsource) [reply pom.xml:159]: Good catch. I've moved the exclusions back.

File: sonar-vbnet-plugin/pom.xml (hunk 2 above)
Comment (costin-zaharia-sonarsource) [reply sonar-vbnet-plugin/pom.xml]: Indeed. I removed all but the one at the root level.

File: pom.xml (hunk 4 above)
Comment (costin-zaharia-sonarsource) [reply pom.xml:122]: `junit.version` was needed but the other no. I've removed them

**Pull request merged**